pkg/apis/core/validation/events.go

```diff
@@ -33,8 +37,85 @@ const (
 	NoteLengthLimit              = 1024
 )
 
-// ValidateEvent makes sure that the event makes sense.
-func ValidateEvent(event *core.Event) field.ErrorList {
+func ValidateEventCreate(event *core.Event, requestVersion schema.GroupVersion) field.ErrorList {
+	// Make sure events always pass legacy validation.
+	allErrs := legacyValidateEvent(event)
+	if requestVersion == core.SchemeGroupVersion || requestVersion == eventsv1beta1.SchemeGroupVersion {
+		// No further validation for backwards compatibility.
+		return allErrs
+	}
+
+	// Strict validation applies to creation via events.k8s.io/v1 API and newer.
+	allErrs = append(allErrs, ValidateObjectMeta(&event.ObjectMeta, true, apimachineryvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, validateV1EventSeries(event)...)
+	zeroTime := time.Time{}
+	if event.EventTime.Time == zeroTime {
+		allErrs = append(allErrs, field.Required(field.NewPath("eventTime"), ""))
+	}
+	if event.Type != v1.EventTypeNormal && event.Type != v1.EventTypeWarning {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("type"), "", fmt.Sprintf("has invalid value: %v", event.Type)))
+	}
+	if event.FirstTimestamp.Time != zeroTime {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("firstTimestamp"), "", "needs to be unset"))
+	}
+	if event.LastTimestamp.Time != zeroTime {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("lastTimestamp"), "", "needs to be unset"))
+	}
+	if event.Count != 0 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("count"), "", "needs to be unset"))
+	}
+	if event.Source.Component != "" || event.Source.Host != "" {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("source"), "", "needs to be unset"))
+	}
+	return allErrs
+}
+
+func ValidateEventUpdate(newEvent, oldEvent *core.Event, requestVersion schema.GroupVersion) field.ErrorList {
+	// Make sure the new event always passes legacy validation.
+	allErrs := legacyValidateEvent(newEvent)
+	if requestVersion == core.SchemeGroupVersion || requestVersion == eventsv1beta1.SchemeGroupVersion {
+		// No further validation for backwards compatibility.
+		return allErrs
+	}
+
+	// Strict validation applies to update via events.k8s.io/v1 API and newer.
+	allErrs = append(allErrs, ValidateObjectMetaUpdate(&newEvent.ObjectMeta, &oldEvent.ObjectMeta, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, validateV1EventSeries(newEvent)...)
```

Comment:
```suggestion
	// if the series was modified, validate the new data
	if !reflect.DeepEqual(newEvent.Series, oldEvent.Series) {
		allErrs = append(allErrs, validateV1EventSeries(newEvent)...)
	}
```

Reply:
only call validateV1EventSeries if the update is changing data; this allows unrelated changes (like removing finalizers, updating labels/annotations) on events persisted with invalid series data via core/v1 or events.k8s.io/v1beta1 APIs

Reply:
Done.